h5c/vic/src/vic-webapp/src/app/create-vch-wizard/compute-capacity/compute-capacity.component.ts

```diff
@@ -178,60 +196,87 @@ export class ComputeCapacityComponent implements OnInit {
-        // value and memory['maxUsage']
-        this.form.get('endpointMemory').setValue(Math.min(memory['maxUsage'], endpointMemoryDefaultValue) + '');
-      } else {
-        this.form.get('endpointMemory').setValidators([]);
-        this.form.get('cpuReservation').setValidators([]);
-        this.form.get('memoryReservation').setValidators([]);
-      }
+        // validates the name of the group and drsStatus
+        this.validateVMGroupsNameAndDrs(this.vmsGroups, drsStatus);
+
+        if (this.inAdvancedMode) {
+          // set max limit validator for endpointMemory
+          this.form.get('endpointMemory').setValidators([
+            ...getNumericValidatorsArray(false),
+            Validators.max(memory['maxUsage'])
+          ]);
+
+          // set max limit validator for cpu unreservedForPool
+          this.form.get('cpuReservation').setValidators([
+            ...getNumericValidatorsArray(false),
+            Validators.max(cpu['unreservedForPool'])
+          ]);
+
+          // set max limit validator for memory unreservedForPool
+          this.form.get('memoryReservation').setValidators([
+            ...getNumericValidatorsArray(false),
+            Validators.max(memory['unreservedForPool'])
+          ]);
+
+          // This prevents the next button from getting disabled when the user selects a host or cluster that has less RAM
+          // available for VM endpoint than the default value. As a solution, we set the smaller value between the default
+          // value and memory['maxUsage']
+          this.form.get('endpointMemory').setValue(Math.min(memory['maxUsage'], endpointMemoryDefaultValue) + '');
+        } else {
+          this.form.get('endpointMemory').setValidators([]);
+          this.form.get('cpuReservation').setValidators([]);
+          this.form.get('memoryReservation').setValidators([]);
+        }
+
+        this.form.get('cpuLimit').updateValueAndValidity();
+        this.form.get('memoryLimit').updateValueAndValidity();
+        this.form.get('endpointMemory').updateValueAndValidity();
+        this.form.get('cpuReservation').updateValueAndValidity();
+        this.form.get('memoryReservation').updateValueAndValidity();
+      });
+  }
 
-      this.form.get('cpuLimit').updateValueAndValidity();
-      this.form.get('memoryLimit').updateValueAndValidity();
-      this.form.get('endpointMemory').updateValueAndValidity();
-      this.form.get('cpuReservation').updateValueAndValidity();
-      this.form.get('memoryReservation').updateValueAndValidity();
-    });
+  /**
+   * Validates if there is a gruop on the selectedResource with the same name as the VCH
+   */
+  private validateVMGroupsNameAndDrs(groups: any[], drsStatus: boolean) {
+    this.vmGroupNameIsValid = !groups['ClusterComputeResource/configurationEx/group'] ||
+      (groups['ClusterComputeResource/configurationEx/group'] && !groups['ClusterComputeResource/configurationEx/group']
```

Comment:
looks like we use 'ClusterComputeResource/configurationEx/group' several times, let's put it into a const